airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -123,103 +131,98 @@ protected void startTracked() throws Exception {
     pairToIgnoredRecordCount.clear();
     LOGGER.info("{} started.", BufferedStreamConsumer.class);
 
-    LOGGER.info("Buffer creation started for {} streams.", pairs.size());
-    final Path queueRoot = Files.createTempDirectory("queues");
-    for (AirbyteStreamNameNamespacePair pair : pairs) {
-      LOGGER.info("Buffer creation for stream {}.", pair);
-      try {
-        final OnDiskQueue writeBuffer = new OnDiskQueue(queueRoot.resolve(pair.getName()), pair.getName());
-        pairToWriteBuffer.put(pair, writeBuffer);
-      } catch (Exception e) {
-        LOGGER.error("Error creating buffer: ", e);
-      }
-    }
-    LOGGER.info("Buffer creation completed.");
-
     onStart.call();
-    LOGGER.info("write buffers: {}", pairToWriteBuffer.keySet());
-    writerPool.scheduleWithFixedDelay(
-        () -> writeStreamsWithNRecords(MIN_RECORDS, pairToWriteBuffer, recordWriter),
-        THREAD_DELAY_MILLIS,
-        THREAD_DELAY_MILLIS,
-        TimeUnit.MILLISECONDS);
   }
 
   @Override
   protected void acceptTracked(AirbyteMessage message) throws Exception {
     Preconditions.checkState(hasStarted, "Cannot accept records until consumer has started");
 
-    if (message.getType() != AirbyteMessage.Type.RECORD) {
-      return;
+    if (message.getType() == Type.RECORD) {
+      final AirbyteRecordMessage recordMessage = message.getRecord();
+      final AirbyteStreamNameNamespacePair stream = AirbyteStreamNameNamespacePair.fromRecordMessage(recordMessage);
+      final String data = Jsons.serialize(message);
+
+      if (!pairs.contains(stream)) {
+        throwUnrecognizedStream(catalog, message);
+      }
+
+      if (!isValidRecord.apply(data)) {
+        pairToIgnoredRecordCount.put(stream, pairToIgnoredRecordCount.getOrDefault(stream, 0L) + 1L);
+        return;
+      }
+
+      if (queue.size() == BATCH_SIZE) {
+        flushQueueToDestination();
+      }
+
+      if (!queue.offer(message)) {
```

Comment:
So we have one global queue now?

Reply:
What is the tradeoff vs the previous implementation?

Reply:
Good question. Keep in mind that the switching to a global queue is a tradeoff is informed by the fact that our state objects are global constrains us to think this way. I'm not changing it because I prefer it for any particular technical or performance reason.

The trade off is that previously we could flush data for any stream to a tmp table independently of any other stream. We were allowed to do this, because the destination did not care about state (we were all or nothing). Because state objects are global, we need to treat the data as if it has a global ordering so that when declare a state message "committed", we know we actually added all of its records. In practice, I do not think this will result in a big performance tradeoff. The worker is already affectively a global queue, we are just pushing that single queue concept a little further into the destination.

There are definitely pathological cases that can make this perform poorly. For example if you had 10K streams and you round robin-ed a record from each stream, it would mean each time we flush data to the temp tables we would be flushing one record to 10K individual tables. I think this pathological case will be rare however. Anecdotally most sources group the records they output by stream already. CDC is a notable exception here, but again, I'm not sure, given the global state object, that there's too much we can do about it. 

If this does become an issue, we can move to a more robust but complex solution where we do per stream queues. The trick here is managing queue size smartly. Likely we would either need to do something clever or go back to spilling to disk, which has its own set of drawbacks.

We can change this approach as we introduce stream-level state objects, but sources that only support global state (e.g. CDC) are going to be constrained to this sort of thinking.